src/strategies/honeyswap/index.ts

```diff
@@ -1,13 +1,129 @@
 import { getAddress } from '@ethersproject/address';
+import { formatUnits } from '@ethersproject/units';
 import { subgraphRequest } from '../../utils';
 
 export const author = 'Kittyslasher';
-export const version = '0.1.0';
+export const version = '0.2.0';
 
 const HONEYSWAP_SUBGRAPH_URL = {
-  '100': 'https://api.thegraph.com/subgraphs/name/1hive/uniswap-v2'
+  '100': {
+    exchange: 'https://api.thegraph.com/subgraphs/name/1hive/uniswap-v2',
+    farm: 'https://api.thegraph.com/subgraphs/name/1hive/honeyfarm-xdai'
+  }
 };
 
+async function getPairs(network, snapshot, token) {
+  const getParams = (prop) => {
+    const params = {
+      pairs: {
+        __args: {
+          where: {
+            [prop]: token.toLowerCase()
+          },
+          first: 1000
+        },
+        id: true,
+        totalSupply: true,
+        reserve0: true,
+        reserve1: true,
+        token0: {
+          id: true,
+          decimals: true
+        },
+        token1: {
+          id: true,
+          decimals: true
+        }
+      }
+    };
+    if (snapshot !== 'latest') {
+      // @ts-ignore
+      params.pairs.__args.block = { number: snapshot };
+    }
+    return params;
+  };
+  const [result1, result2] = await Promise.all(
+    ['token0', 'token1'].map((prop) =>
+      subgraphRequest(HONEYSWAP_SUBGRAPH_URL[network].exchange, getParams(prop))
+    )
+  );
+  return Object.fromEntries(
+    result1.pairs.concat(result2.pairs).map((pair) => {
+      const isToken0 = pair.token0.id == token.toLowerCase();
+      const rate = isToken0
+        ? +pair.reserve0 / +pair.totalSupply
+        : +pair.reserve1 / +pair.totalSupply;
+      const decimals = isToken0 ? pair.token0.decimals : pair.token1.decimals;
+      return [pair.id, { rate, decimals }];
+    })
+  );
+}
+
+async function getPools(network, snapshot, token) {
+  const pairs = await getPairs(network, snapshot, token);
+  const params = {
+    pools: {
+      __args: {
+        where: {
+          id_in: Object.keys(pairs)
+        },
+        first: 1000
+      },
+      id: true
+    }
+  };
+  if (snapshot !== 'latest') {
+    // @ts-ignore
+    params.pools.__args.block = { number: snapshot };
+  }
+  const result = await subgraphRequest(
+    HONEYSWAP_SUBGRAPH_URL[network].farm,
+    params
+  );
+  return Object.fromEntries(
+    result.pools.map((pool) => [pool.id, pairs[pool.id]])
+  );
+}
+
+async function getStakedBalances(network, snapshot, token, addresses) {
+  const pools = await getPools(network, snapshot, token);
+  const params = {
+    deposits: {
+      __args: {
+        where: {
+          pool_in: Object.keys(pools),
+          user_in: addresses.map((addr) => addr.toLowerCase())
```

Comment:
This doesn't scale well, the array on arguments have limit, I don't remember how many but call will fail at some point, maybe after 1000 votes. You may need to find another way to gather this information if you expect more vote than this